x/emissions/keeper/msgserver/msg_server_stake.go

```diff
@@ -57,34 +57,29 @@ func (ms msgServer) AddStake(ctx context.Context, msg *types.MsgAddStake) (*type
 // if a stake removal is not confirmed within a certain time period, the stake removal becomes invalid
 // and one must start the stake removal process again and wait the delay again.
 func (ms msgServer) StartRemoveStake(ctx context.Context, msg *types.MsgStartRemoveStake) (*types.MsgStartRemoveStakeResponse, error) {
-	sdkCtx := sdk.UnwrapSDKContext(ctx)
 	// Check the sender is registered
-	senderAddr, err := sdk.AccAddressFromBech32(msg.Sender)
+	sender := sdk.AccAddress(msg.Sender)
+	// Check the sender has enough stake already placed on the topic to remove the stake
+	stakePlaced, err := ms.k.GetStakeOnTopicFromReputer(ctx, msg.TopicId, sender)
 	if err != nil {
 		return nil, err
 	}
-	stakeRemoval := types.StakeRemoval{
-		BlockRemovalStarted: sdkCtx.BlockHeight(),
-		Placements:          make([]*types.StakePlacement, 0),
+	if stakePlaced.LT(msg.Amount) {
+		return nil, types.ErrInsufficientStakeToRemove
 	}
-	for _, stakePlacement := range msg.PlacementsRemove {
-		// Check the sender has enough stake already placed on the topic to remove the stake
-		stakePlaced, err := ms.k.GetStakeOnTopicFromReputer(ctx, stakePlacement.TopicId, senderAddr)
-		if err != nil {
-			return nil, err
-		}
-		if stakePlaced.LT(stakePlacement.Amount) {
-			return nil, types.ErrInsufficientStakeToRemove
-		}
 
-		// Push to the stake removal object
-		stakeRemoval.Placements = append(stakeRemoval.Placements, &types.StakePlacement{
-			TopicId: stakePlacement.TopicId,
-			Amount:  stakePlacement.Amount,
-		})
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	stakeToRemove := types.StakeRemoval{
```

Comment:
wait you can only remove one stake at a time?

Reply:
like what if someone wants to just unstake everywhere?

Reply:
gas is cheap here, should be fine. I don't imagine people to be participating in many topics as each requires specialized knowledge